### PR TITLE
Distributor config added to publish history

### PR DESCRIPTION
**Pull request by midnightercz**
It could be handy to have distributor config included in each entry of publish history

### CODE REVIEW

**Comment by ipanova** [comment]
this PR is related to https://pulp.plan.io/issues/973

File: server/pulp/server/managers/repo/publish.py [hunk 1]
```diff
@@ -165,7 +165,9 @@ def _do_publish(repo, distributor_id, distributor_instance, transfer_repo, condu
             result_code = RepoPublishResult.RESULT_SUCCESS
 
         result = RepoPublishResult.expected_result(
-            repo_id, repo_distributor['id'], repo_distributor['distributor_type_id'],
+            repo_id, repo_distributor['id'],
+            call_config.flatten(),
+            repo_distributor['distributor_type_id'], call_config.flatten(),
             publish_start_timestamp, publish_end_timestamp, summary, details, result_code)
```
Comment (rbarlow) [inline server/pulp/server/managers/repo/publish.py]: The above doesn't seem to match the call arguments:

```
def expected_result(cls, repo_id, distributor_id, distributor_type_id,
                        distributor_config, started, completed, summary,
                        details, result_code):
```

File: server/test/unit/server/managers/repo/test_publish.py [hunk 2]
```diff
@@ -606,6 +606,30 @@ def test_publish_history_missing_repo(self):
...
+        """
+        Test there's distributor config included in publish history
+        """
+        # Setup
+        self.repo_manager.create_repo('test_date')
+        self.distributor_manager.add_distributor('test_date', 'mock-distributor', {}, True,
+                                                 distributor_id='test_dist')
+        add_result('test_date', 'test_dist', 1)
+        entry = self.publish_manager.publish_history('test_date', 'test_dist')[0]
+        self.assertTrue(hasattr(entry, "distributor_config"))
```
Comment (rbarlow) [inline server/test/unit/server/managers/repo/test_publish.py]: This test should assert more than just the existence of an attribute called distributor_config - it should verify that the data in the distributor_config is corect.

File: server/test/unit/server/managers/repo/test_publish.py [hunk 3]
```diff
@@ -606,6 +606,30 @@ def test_publish_history_missing_repo(self):
...
+        """
+        Test there's distributor config included in publish history
+        """
+        # Setup
+        self.repo_manager.create_repo('test_date')
+        self.distributor_manager.add_distributor('test_date', 'mock-distributor', {}, True,
+                                                 distributor_id='test_dist')
+        add_result('test_date', 'test_dist', 1)
+        entry = self.publish_manager.publish_history('test_date', 'test_dist')[0]
+        self.assertTrue("distributor_config" in entry, entry)
```
Comment (rbarlow) [inline server/test/unit/server/managers/repo/test_publish.py]: This test has the same name as the one above it, which is an error. Also, the code is almost identical except the assertion is slightly different.

File: server/pulp/server/db/model/repository.py [hunk 4]
```diff
@@ -427,8 +427,9 @@ def error_result(cls, repo_id, distributor_id, distributor_type_id, started, com
         return r
 
     @classmethod
-    def expected_result(cls, repo_id, distributor_id, distributor_type_id, started,
-                        completed, summary, details, result_code):
+    def expected_result(cls, repo_id, distributor_id, distributor_type_id,
+                        distributor_config, started, completed, summary,
+                        details, result_code):
```
Comment (rbarlow) [inline server/pulp/server/db/model/repository.py]: There doesn't seem to be a test that explicitly tests this change.

File: server/test/unit/server/managers/repo/test_publish.py [hunk 5]
```diff
@@ -482,7 +482,7 @@ def test_publish_history_ascending_sort(self):
         date_string = '2013-06-01T12:00:0%sZ'
         for i in range(0, 10, 2):
             r = RepoPublishResult.expected_result(
-                'test_sort', 'test_dist', 'bar', date_string % str(i), date_string % str(i + 1),
+                'test_sort', 'test_dist', 'bar', {}, date_string % str(i), date_string % str(i + 1),
```
Comment (rbarlow) [inline server/test/unit/server/managers/repo/test_publish.py]: This and many other tests have been updated to pass an empty dictionary here. Please make a new test that passes some data to this method and asserts that expected_result() did the right thing with the information.

File: server/pulp/server/db/model/repository.py [hunk 6]
```diff
@@ -508,6 +511,7 @@ def __init__(self, repo_id, distributor_id, distributor_type_id, started, comple
         self.repo_id = repo_id
         self.distributor_id = distributor_id
         self.distributor_type_id = distributor_type_id
+        self.distributor_config = distributor_config
```
Comment (rbarlow) [inline server/pulp/server/db/model/repository.py]: Please create a test that specifically asserts that this __init__() method sets this attribute.

File: server/pulp/server/managers/repo/publish.py (hunk 1 above)
Comment (rbarlow) [inline server/pulp/server/managers/repo/publish.py]: Please make sure that the above is explicitly tested so we can be sure that it works properly.

**Comment by rbarlow** [comment]
By policy, we need these changes to be made in a single commit. We can't merge this with 2.7-dev, because it is a behavioral change. I'm also concerned that the tests are not making strong assertions about the data being stored, and that the call to that function does not match the method signature.

I'm going to close this since it needs to be changed in the master branch instead of 2.7-dev. Here are the things we need fixed about this change before we can merge to master:

0) Make the change in a single commit.
1) Have the commit message reference the Redmine issue with "fixes #973" on its own line.
2) Have the tests make strong assertions about the distributor_config, not just the attribute existing.
3) Make sure there are no duplicate tests.
4) Make sure the call parameters are used correctly.
5) Make sure there is a test for each item that you've changed, asserting that the new behavior is working correctly.

**Pull request closed without merging**